chainerx_cc/chainerx/routines/manipulation.cc

```diff
@@ -719,4 +719,33 @@ Array Swapaxes(const Array& a, int8_t axis1, int8_t axis2) {
     return out;
 }
 
+Array ExpandDims(const Array& a, int8_t axis) {
+    Shape shape = a.shape();
+    Strides strides = a.strides();
+
+    auto ndim = a.ndim();
+    if (axis < -ndim - 1 || ndim < axis) {
+        throw DimensionError{"Axis ", axis, " is out of bounds for array of dimension ", ndim};
+    }
+
+    if (axis < 0) {
+        axis = axis + ndim + 1;
+    }
+
+    shape.insert(shape.begin() + axis, 1);
+
+    Array out = a.Reshape(shape);
+
+    BackwardBuilder bb{"expand_dims", a, out};
+    if (BackwardBuilder::Target bt = bb.CreateTarget(0)) {
+        bt.Define([axis](BackwardContext& bctx) {
+            const Array& gout = *bctx.output_grad();
+            bctx.input_grad() = gout.Squeeze(Axes{axis});
+        });
+    }
+    bb.Finalize();
```

Comment:
This backward definition is not needed because `Reshape` at [L737](https://github.com/chainer/chainer/pull/7029/files#diff-1a0097987b5e6a8fe06006b09904a09fR737) includes it.

Reply:
My bad. Removed that